okonomiyaki/file_formats/_egg_info.py

```diff
@@ -820,6 +881,22 @@ def egg_name(self):
         return self._spec_depend.egg_name
 
     @property
+    def is_strictly_supported(self):
+        """ Returns True if the given metadata_version is fully supported.
+
+        A metadata_version is fully supported iff:
+            - metadata_version.major ==
+              EggMetadata.HIGHEST_SUPPORTED_METADATA_VERSION.major
+            - and metadata_version.minor <=
+              EggMetadata.HIGHEST_SUPPORTED_METADATA_VERSION.minor
+        """
+        max_supported = EggMetadata.HIGHEST_SUPPORTED_METADATA_VERSION
+        return (
+            _are_compatible(self.metadata_version, max_supported) and
```

Comment:
I expect to support `2.0` w/ an API similar to what I did for runtimes: one class per major version, and some factory that would automatically return an instance of the right class.